snapcraft/tests/test_commands_help.py

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright (C) 2015 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import builtins
+import importlib
+import io
+import fixtures
+
+from unittest import mock
+
+# Snapcraft modules
+from snapcraft import tests
+from snapcraft.commands import help
+
+
+class HelpCommandTestCase(tests.TestCase):
+
+    def test_topic_and_plugin_not_found_exists_with_tip(self):
+        fake_logger = fixtures.FakeLogger()
+        self.useFixture(fake_logger)
+
+        with self.assertRaises(SystemExit) as raised:
+            help.main(['does-not-exist'])
+
+        self.assertEqual(raised.exception.code, 1, 'Wrong exit code returned.')
+        self.assertEqual('The plugin does not exist. Run `snapcraft '
+                         'list-plugins` to see the available plugins.\n',
+                         fake_logger.output)
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    def test_print_module_help_when_no_help_for_valid_plugin(
+            self, mock_stdout):
+        help.main(['jdk'])
+
+        self.assertEqual('The plugin has no documentation\n',
+                         mock_stdout.getvalue())
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    def test_print_module_help_for_valid_plugin(self, mock_stdout):
+        help.main(['nil'])
+
+        output = mock_stdout.getvalue()
+        self.assertTrue(output.startswith('The nil plugin is'),
+                        'Help starts with {!r}'.format(output))
+
+    @mock.patch.object(builtins, 'help')
+    def test_show_module_help_with_devel_for_valid_plugin(self, mock_help):
+        help.main(['nil', '--devel'])
+
+        module = importlib.import_module('snapcraft.plugins.nil')
+        mock_help.assert_called_once_with(module)
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    def test_print_topics(self, mock_stdout):
+        help.main(['topics'])
+
+        output = mock_stdout.getvalue().strip().split('\n')
+        for t in help._TOPICS:
+            self.assertTrue(
+                t in output,
+                'Missing topic: {!r} in {}'.format(t, output))
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    def test_print_topic_help_for_valid_topic(self, mock_stdout):
+        help.main(['sources'])
+
+        output = mock_stdout.getvalue()
+        self.assertTrue(output.startswith('Common keywords for plugins'),
+                        'Help starts with {!r}'.format(output))
```

Comment:
Won't this error actually print the entire output rather than what it "starts with?" That may not be the most helpful test failure if someone just changed the first few words. What if you read until newline and validated against "Common keywords for plugins that use common source options", and made the error "Expected the first sentence to be X, but it was Y?"